### PR TITLE
MNT Ignore some additional failures

### PR DESCRIPTION
**Pull request by emteknetnz**
Also merge these PRs to remove JS extensions:

Passing
- [ ] https://github.com/silverstripe/silverstripe-asset-admin/pull/1238
- [ ] https://github.com/silverstripe/silverstripe-campaign-admin/pull/212
- [ ] https://github.com/silverstripe/silverstripe-ckan-registry/pull/256
- [ ] https://github.com/silverstripe/silverstripe-cms/pull/2700
- [ ] https://github.com/silverstripe/silverstripe-tagfield/pull/196
- [ ] https://github.com/silverstripe/silverstripe-admin/pull/1262

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -41,6 +41,15 @@ module.exports = {
   'rules': Object.assign({},
     todo,
     {
+      // turned off otherwise non-admin modules will complain about importing components from admin
+      // via the novel silverstripe js component sharing setup
+      'import/no-extraneous-dependencies': [
```
Comment (maxime-rainville) [inline .eslintrc.js:46]: That bit is fine. I was getting that on some of my personal project as well, was wondering where in was coming from.

File: .eslintrc.js [hunk 2]
```diff
@@ -41,6 +41,15 @@ module.exports = {
...
+        'off'
+      ],
+      // turned off to allow the using of the js file extension for /boot/index.js
+      'import/extensions': [
```
Comment (maxime-rainville) [inline .eslintrc.js:50]: Couldn't we just remove the file extension in those imports? I would rather stay as close as possible to the base `eslint-config-airbnb` rule set.

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
Update the `version` attribute in `package.json` to `0.0.6` so we can tag a new release in NPM.

File: .eslintrc.js (hunk 2 above)
Comment (emteknetnz) [reply .eslintrc.js:50]: Means we'd need to update all the affected modules individually e.g. https://github.com/silverstripe/silverstripe-campaign-admin/blob/1/client/src/bundles/bundle.js and however many else

We'd also need to do this for older versions too such as 1.9.  Massive pain :-/

Comment (emteknetnz) [reply .eslintrc.js:50]: There's a large number of files affected, doing a regex search `require\(.+?\.js'` on a sink install netted 47 results in 21 files

A few random examples:
- https://github.com/silverstripe/silverstripe-admin/blob/1/client/src/legacy/SecurityAdmin.js#L6
- https://github.com/silverstripe/silverstripe-ckan-registry/blob/1/client/src/bundles/bundle.js#L2
- https://github.com/silverstripe/silverstripe-admin/blob/1/client/src/legacy/ssui.core.js#L3

I really don't think all the pull-requests required to fix this justifies the trivial linting benefit

Comment (maxime-rainville) [reply .eslintrc.js:50]: If every time we get linting errors, we disable our linting rules, we might as well not have any linting rules at all. As far I'm concern, having a rule to disallow adding useless `.js` extension in require statements is perfectly sensible.

If we don't have the time to fix these linting errors right now, lets change the constraint in `package.json` so we get the older version of `eslint-config-airbnb` that allow `require` statements. But we'll have to get to it eventually.

**Review by maxime-rainville — CHANGES_REQUESTED** [review]

**Comment by emteknetnz** [comment]
> lets change the constraint in package.json

That's far worse, that would require updating literally every module that uses js

> If every time we get linting errors, we disable our linting rules, we might as well not have any linting rules at all

We're already ignoring a large number of rules
https://github.com/silverstripe/eslint-config/blob/master/.eslintrc.js

This is honestly no different from us ignoring a bunch of PSR2 rules
https://github.com/silverstripe/silverstripe-framework/blob/4/phpcs.xml.dist#L11

I honestly don't see what the issue is here.  I don't see any value is doing a bunch of work to fix such a trivial issue

**Comment by maxime-rainville** [comment]
We should only ignore rules if there's something about our peculiar set up that make it impossible or impractical to follow said rule. The rule we are debating here is totally easy to follow within our current set up and we are already following it for the most part.

I'm happy to help out upgrading the other PRs if it make it easier.

**Comment by emteknetnz** [comment]
Looks like the `.js` file extensions are required - the following PR caused the following JS job to fail
https://github.com/silverstripe/silverstripe-admin/pull/1262/files#diff-2bd1317ee649ea721f5e3d72a7d3f6bdd38f46e81d65dfedb3f9959e870229deR14
https://app.travis-ci.com/github/silverstripe/silverstripe-admin/jobs/546228067

File: .eslintrc.js (hunk 2 above)
Comment (michalkleiner) [reply .eslintrc.js:50]: Agree with Max 👍

**Comment by emteknetnz** [comment]
@maxime-rainville I've removed the bit where JS extensions are allowed, so they're once again disallowed

**Comment by maxime-rainville** [comment]
If all the problems are in `bundle.js` and `vendor.js`, I would rather we disabled those linting rule for those files only. Those files are weird and there's already a recognition of their weirdness in the fact that [we've already disabled a bunch of linting warnings](https://github.com/silverstripe/silverstripe-admin/blob/a7e6d13f2adfb23cf58613c363497d2b448c799d/client/src/bundles/bundle.js#L1-L5) for them only.

```js
/* eslint-disable
 import/no-webpack-loader-syntax,
 import/no-extraneous-dependencies,
 import/no-unresolved
 */
```

**Comment by emteknetnz** [comment]
The issue isn't that linting, it's that javascript doesn't actually work when the .js file extensions are not there

**Comment by emteknetnz** [comment]
I've removed the ignore-rule from eslint and add the ignore annotation to admin

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
Still need to do this.

> Update the version attribute in package.json to 0.0.6 so we can tag a new release in NPM.

**Comment by emteknetnz** [comment]
> 0.0.6
done

**Review by maxime-rainville — APPROVED** [review]

**Pull request merged**